src/shared/rm-rf.h

```diff
@@ -51,3 +52,19 @@ static inline char *rm_rf_subvolume_and_free(char *p) {
         return mfree(p);
 }
 DEFINE_TRIVIAL_CLEANUP_FUNC(char*, rm_rf_subvolume_and_free);
+
+static inline int rm_rf_physical_and_close(int fd) {
+        _cleanup_free_ char *p = NULL;
+
+        if (fd < 0)
+                return -1;
+
+        if (fd_get_path(fd, &p) < 0)
+                return safe_close(fd);
+
+        safe_close(fd);
+        (void) rm_rf(p, REMOVE_ROOT|REMOVE_PHYSICAL|REMOVE_MISSING_OK|REMOVE_CHMOD);
```

Comment:
hmpf, i don't really like this I must say fd_get_path() is entirely unreliable in particular for stuff that has already been deleted (as the path it reports will then be suffixed with `(removed)`, and thus is entirely ambigious). 

Sorry, but this cannot work I fear.

fd_get_path() is OK for some things (informational stuff for example), but I think using it for destructive file operations is a big problem.

This goes particularly bad if the fd passed refers to something that is not actually a thing in the file system (sockets, eventfd, …).

I#d really avoid this.

Reply:
Dropped this stuff again